src/compiler/diagnosticMessages.json

```diff
@@ -127,11 +127,27 @@
         "category": "Error",
         "code": 1039
     },
+    "'{0}' modifier cannot be used in an ambient context.": {
+        "category": "Error",
+        "code": 1040
+    },
+    "'{0}' modifier cannot be used with a class declaration.": {
+        "category": "Error",
+        "code": 1041
+    },
+    "'{0}' modifier cannot be used here.": {
+        "category": "Error",
+        "code": 1042
+    },
+    "'{0}' modifier cannot appear on a data property.": {

```

Comment:
I may need a better term here. In TypeScript we use the terms “property”, “method”, and “accessor”. In JavaScript, they use the terms “data property” and “accessor property”. A “data property” is a property whose descriptor has a “value”, while an “accessor property” is a property whose descriptor has a “get” and/or “set”.

From: Jason Freeman [mailto:notifications@github.com]
Sent: Thursday, February 12, 2015 2:25 PM
To: Microsoft/TypeScript
Cc: Ron Buckton
Subject: Re: [TypeScript] Async Functions (#1781)


In src/compiler/diagnosticMessages.json<https://github.com/Microsoft/TypeScript/pull/1781#discussion_r24626742>:

> @@ -127,11 +127,27 @@

>          "category": "Error",

>          "code": 1039

>      },

> +    "'{0}' modifier cannot be used in an ambient context.": {

> +        "category": "Error",

> +        "code": 1040

> +    },

> +    "'{0}' modifier cannot be used with a class declaration.": {

> +        "category": "Error",

> +        "code": 1041

> +    },

> +    "'{0}' modifier cannot be used here.": {

> +        "category": "Error",

> +        "code": 1042

> +    },

> +    "'{0}' modifier cannot appear on a data property.": {

What's a data property?

—
Reply to this email directly or view it on GitHub<https://github.com/Microsoft/TypeScript/pull/1781/files#r24626742>.